weed/storage/volume_vacuum.go

```diff
@@ -327,7 +341,11 @@ func (v *Volume) makeupDiff(newDatFileName, newIdxFileName, oldDatFileName, oldI
 		}
 	}
 
-	return nil
+	if v.needleMapKind == NeedleMapInMemory {
```

Comment:
why this is needed if v.tmpNm already has different implementations?

Reply:
We need to do twice DoOffsetLoading:
1) Loading the  new .idx to tmpNm when copyDataAndGenerateIndexFile. At the same time new updated or deleted files may be written to the same volume.
2)  Loading the new .idx files from offset to the tmpNm. The offset is the start for .idx of new files when doing copyDataAndGenerateIndexFile.

After 1 and 2 , an intact needlemap can be generated.

Reply:
With the different implementation for the temp mapper interface, the code should not need this `if v.needleMapKind == NeedleMapInMemory` checking for different leveldb and memory implementation.

Reply:
Yes, this check is not needed. At first I misunderstand your meaning.
I forgot to remove the check after I used  interface instead of  reflactor.

Reply:
Done